pilot/pkg/model/context.go

```diff
@@ -277,10 +267,27 @@ func GetProxyConfigNamespace(proxy *Proxy) string {
 	}
 
 	// First look for ISTIO_META_CONFIG_NAMESPACE
+	// All newer proxies (from Istio 1.1 onwards) are supposed to supply this
 	if configNamespace, found := proxy.Metadata["CONFIG_NAMESPACE"]; found {
 		return configNamespace
 	}
 
+	// if not found, for backward compatibility, extract the namespace from
+	// the proxy domain. this is a k8s specific hack and should be enabled
+	// only for 1.0.x proxies
+	if version, found := proxy.Metadata["ISTIO_PROXY_VERSION"]; found {
```

Comment:
Version is not reliably included - why do we need this ? If CONFIG_NAMESPACE is missing - just get it from the DNSDomain.

Reply:
I mean - not having CONFIG_NAMESPACE is enough indication that the namespace should be extracted from node id - no need for the rest of the checks and code.

Reply:
What do you mean version is not reliably included? we have had the version since 1.0.0 release. Atleast this way, we restrict the dns parsing hack to just 1.0 proxies and do not extend this infinitely to all proxies in all releases in every platform. If there is no config namespace, you should be returning "" as the namespace and not one thats arbitrary parsed from the DNS value. CloudFoundry for instance supplies sidecar~someIP~X~X where its literally X, X. There are also Consul deployments that supply dummy parameters. Simply parsing the DNs value would automatically force all these systems to some arbitrary namespace that users would have to keep track of, and use in their configs.

Reply:
The 1.1 proxies will have CONFIG_NAMESPACE. If a proxy doesn't have CONFIG_NAMESPACE - it is clearly not a 1.1 proxy. So no need to add extra version checks.

In 1.0 we failed to add VERSION to mesh expansion, so the meta is missing from the release.